File: frontend/src/services/shopping.ts
// Shopping recommendations service
import axios from 'axios';

const API_URL = 'http://localhost:5001/api';

export interface ShoppingRecommendation {
  store: string;
  emoji: string;
  url: string;
  query: string;
}

export interface ShoppingResponse {
  success: boolean;
  search_query: string;
  recommendations: ShoppingRecommendation[];
  error?: string;
}

export const getShoppingRecommendations = async (analysis: any): Promise<ShoppingResponse> => {
  const response = await axios.post(`${API_URL}/shopping/recommendations`, {
    analysis
  });
  return response.data;
};

File: backend-python/app/api/shopping.py
from flask import Blueprint, request, jsonify
from app.services.shopping_service import shopping_service

shopping_bp = Blueprint(
    "shopping",
    __name__,
    url_prefix="/api/shopping"
)

@shopping_bp.route("/recommendations", methods=["POST"])
def get_shopping_recommendations():
    data = request.get_json()

    if not data or "analysis" not in data:
        return jsonify({
            "success": False,
            "error": "Missing analysis data"
        }), 400

    analysis = data["analysis"]

    search_query = shopping_service.generate_search_query(analysis)
    recommendations = shopping_service.generate_shopping_links(analysis)

    return jsonify({
        "success": True,
        "search_query": search_query,
        "recommendations": recommendations
    }), 200
